tests/unit/fem/test_pa_kernels.cpp

```diff
@@ -445,7 +454,33 @@ TEST_CASE("PA Convection", "[PartialAssembly][MFEMData]")
    {
       test_pa_convection("../../data/periodic-square.mesh", order_2d, prob,
                          refinement_2d);
-      if (launch_all_non_regression_tests)
+   }
+
+   SECTION("3D")
+   {
+      test_pa_convection("../../data/periodic-cube.mesh", order_3d, prob,
+                         refinement_3d);
+   }
+} // test case
+
+// Advanced unit tests for convection
+TEST_CASE("PA Convection advanced", "[PartialAssembly][MFEMData]")
+{
+   if (launch_all_non_regression_tests)
+   {
+      // prob:
+      // - 0: CG,
+      // - 1: DG continuous coeff,
+      // - 2: DG discontinuous coeff,
+      // - 3: DG Bernstein discontinuous coeff.
+      auto prob = GENERATE(0, 1, 2, 3);
+      auto order_2d = GENERATE(2);
+      auto order_3d = GENERATE(2);
+      // refinement > 0 => Non-conforming mesh
+      auto refinement_2d = GENERATE(0,1);
+      auto refinement_3d = GENERATE(0,1);
```

Comment:
You should put the 2D and 3D `GENERATE` inside the respective `SECTION`s to avoid running duplicate tests

Reply:
Similarly for the `"PA Convection"` test case

Reply:
done in [45ac641](https://github.com/mfem/mfem/pull/2840/commits/45ac64129e74798ea90c9964717adfbe267bcb8c)

Reply:
I can't believe I missed it right after explaining it...